autopilot/hostscore.go

```diff
@@ -243,3 +276,93 @@ func randSelectByWeight(weights []float64) int {
 	}
 	return len(weights) - 1
 }
+
+// contractPriceForScore returns the contract price of the host used for
+// scoring. Since we don't know whether rhpv2 or rhpv3 are used, we return the
+// bigger one for a pesimistic score.
+func contractPriceForScore(h hostdb.Host) types.Currency {
+	cp := h.Settings.ContractPrice
+	if cp.Cmp(h.PriceTable.ContractPrice) > 0 {
+		cp = h.PriceTable.ContractPrice
+	}
+	return cp
+}
+
+func uploadCostForScore(cfg api.AutopilotConfig, h hostdb.Host, bytes uint64) types.Currency {
+	uploadSectorCostRHPv2, _ := rhpv2.RPCAppendCost(*h.Settings, cfg.Contracts.Period)
+
+	asc := h.PriceTable.AppendSectorCost(cfg.Contracts.Period)
+	uploadSectorCostRHPv3, _ := asc.Total()
+
+	numSectors := bytes / rhpv2.SectorSize
+	if bytes%rhpv2.SectorSize != 0 {
+		numSectors++
+	}
+	if uploadSectorCostRHPv2.Cmp(uploadSectorCostRHPv3) > 0 {
+		return uploadSectorCostRHPv2.Mul64(numSectors)
+	}
+	return uploadSectorCostRHPv3.Mul64(numSectors)
+}
+
+func downloadCostForScore(cfg api.AutopilotConfig, h hostdb.Host, bytes uint64) types.Currency {
+	downloadSectorCostRHPv2 := rhpv2.RPCReadCost(*h.Settings, []rhpv2.RPCReadRequestSection{{Offset: 0, Length: rhpv2.SectorSize}})
+	asc := h.PriceTable.ReadSectorCost(rhpv2.SectorSize)
+	downloadSectorCostRHPv3, _ := asc.Total()
+
+	numSectors := bytes / rhpv2.SectorSize
+	if bytes%rhpv2.SectorSize != 0 {
+		numSectors++
+	}
+	if downloadSectorCostRHPv2.Cmp(downloadSectorCostRHPv3) > 0 {
+		return downloadSectorCostRHPv2.Mul64(numSectors)
+	}
+	return downloadSectorCostRHPv3.Mul64(numSectors)
+}
+
+func storageCostForScore(cfg api.AutopilotConfig, h hostdb.Host, bytes uint64) types.Currency {
+	storeSectorCostRHPv2 := h.Settings.StoragePrice.Mul64(bytes)
+
+	asc := h.PriceTable.AppendSectorCost(cfg.Contracts.Period)
+	storeSectorCostRHPv3 := asc.Storage
+
+	numSectors := bytes / rhpv2.SectorSize
+	if bytes%rhpv2.SectorSize != 0 {
+		numSectors++
+	}
+	if storeSectorCostRHPv2.Cmp(storeSectorCostRHPv3) > 0 {
+		return storeSectorCostRHPv2.Mul64(numSectors)
+	}
+	return storeSectorCostRHPv3.Mul64(numSectors)
+}
+
+func hostPeriodCostForScore(h hostdb.Host, cfg api.AutopilotConfig, expectedRedundancy float64) types.Currency {
+	// TODO: if we ever stop assuming that certain prices in the pricetable are
```

Comment:
we should probably add this as a `NOTE` in the gouging code right, because we make those assumptions there